src/View/Widget/DateTimeWidget.php

```diff
@@ -218,11 +218,11 @@ protected function _deconstructDate($value, $options)
             ];
         }
         try {
-            if (is_string($value)) {
+            if (is_string($value) && !is_numeric($value)) {
                 $date = new DateTime($value);
             } elseif (is_bool($value)) {
                 $date = new DateTime();
-            } elseif (is_int($value)) {
+            } elseif (is_int($value) || is_numeric($value)) {
                 $date = new DateTime('@' . $value);
```

Comment:
Timestamps are always in UTC. When using passing a timestamp as first param to `DateTime` the second param is ignored even if passed.

Quoting the manual:

> The $timezone parameter and the current timezone are ignored when the $time parameter either is a UNIX timestamp (e.g. @946684800) or specifies a timezone (e.g. 2010-01-28T15:00:00+02:00).